js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties

```diff
@@ -2456,7 +2456,7 @@ targetContextAttributes=Target Context Attributes
 targetContextAttributesHelp=Defines the evaluation of context attributes (claims) instead of identity attributes
 filteredByClaim=Verify essential claim
 rowCancelBtnAriaLabel=Cancel edits for {{messageBundle}}
-validateSignatureHelp=Enable/disable signature validation of external IDP signatures.
+validateSignatureHelp=If enabled, the JWT signature will be validated using the keys configured below. For the JWT Authorization Grant, the signature is always validated even if this switch is disabled.
```

Comment:
@graziang I think that this is not OK. Right now the `validateAuthorizationGrantAssertion` calls the `verify` method, and this one checks the property to see if the validation is enabled or not ([here](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java#L632)). I suppose we want to move that check out the method or create a new method without that check. This is also the same for spiffe ([here](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java#L1057)). Don't know if we want to do the same for spiffe...

Reply:
@rmartinc @graziang +1 for this point. 

If jwt-bearer grant is used, we must always validate the signature on the JWT assertion. There should never be an option to allow jwt-bearer grant without the signature validation as this would be very unsecure. That is also my understanding per the specs https://datatracker.ietf.org/doc/html/rfc7523#section-3 . I think that same requirement applies for spiffe as well.

I can see that for spiffe (method `OIDCIdentityProvider.verifyClientAssertion`) there is required that OIDC provider must have `validateSignatures` enabled. Maybe we can do the same for jwt-bearer grant as well? That would be easier as we will not need to change anything in the UI and keep the UI as is in current Keycloak main.

Reply:
We have been chatting about this in slack. Yes, we concluded the same, trying to do the same that Spiffe. If validate signature is not enabled throw an exception, saying that signature validation is not enabled or similar. But maybe hiding in the admin console the JWT section and the `supports client assertions` for Spiffe when signature is disabled is better. This way it's clear that with no signature validation, no JWT or Spiffe.